R/plotSeries.R

```diff
@@ -0,0 +1,147 @@
+                   x_axis = NULL,
+                   y_axis = NULL,
+                   rank = NULL){
+              
+              ###################### Input check #######################
+              # Check abund_values
+              .check_abund_values(abund_values, x)
+              
+              # Check x_axis
+              if( !(x_axis %in% names(colData(x))) ){
+                  stop("'x_axis' must be a name of column of colData(x)", call. = FALSE)
+              }
+              
+              # Check y_axis
+              # If y_axis is not null, user has specified it
+              if (!is.null(y_axis)){
+                  if(!(y_axis %in% rownames(x)) ){
+                      stop("'y_axis' must be in rownames(x).", call. = FALSE)
+                  }
+              }
+              
+              
+              # Check rank
+              # IF rank is not null, it has been specified by user
+              if( !is.null(rank) ){
+                  if(!.is_non_empty_string(rank)){
+                      stop("'rank' must be an non empty single character value,
+                           and it must be one of taxonomyRanks(x).",
+                           call. = FALSE)
+                  }
+                  .check_taxonomic_rank(rank, x)
+              }
+              ##############################################
+              
+              # If rank is not null, data will be agglomerated by rank
+              if( !is.null(rank) ){
+                  x <- agglomerateByRank(x, rank = rank)
+                  
+                  # Rownames are now changed, so y-axis must be checked again
+                  # Check y_axis
+                  if (!is.null(y_axis)){
+                      if(!(y_axis %in% rownames(x)) ){
+                          stop("'y-axis' does not match to rownames(x). 
+                          Data is agglomerated by 'rank', and rownames(x) are
+                          also changed in relation to 'rank'. Check that 'y-axis'
+                          match to agglomerated data.", call. = FALSE)
+                      }
+                  }
+              }
+              
+              # If y_axis is not null, take only those taxa into account that user
+              # is specified
+              if( !is.null(y_axis) ){
+                  x <- x[y_axis]
+              }
+              
+              # Melt data
+              melted_data <- meltAssay(x, add_col_data = TRUE, add_row_data = TRUE)
+              color_by = "Kingdom"
+              line_by = "FeatureID"
+              
+              # # Take only those columns that are needed
+              # melted_data <- melted_data[, c("FeatureID", abund_values, x_axis)]
+              
+              ggplot2::ggplot(melted_data) + 
+                  ggplot2::geom_line(
+                      ggplot2::aes_string(x = x_axis, y = abund_values, 
+                                          color = color_by, linetype = line_by))
```

Comment:
Use the same function structure as used in the other plotting functions.

1. `.get_assay_data` should include retrieval of assay data. Include a check/warning for to much data. Maybe > 10 should get a warning and > 20 should get an error, since this will in every event not make sense. 
2. `.incorporate_vis` should include the incorporation of series information based on the colData, and colour, linetype, size parameter based on rowData. The additional columns should be named as the arguments, e.g. `X`, `colour_by`, `size_by`, `linetype_by`. `cbind` it to output of 1 and make sure to play back the found names, since partial name matching is supported by `retrieveCellInfo` and `retrieveFeatureInfo`.
3. `.melt_series_data` should include the melting step using `pivot_longer`. Here the series information has to be returned as `X` (capital letter) and the different assay values as `Y`. the rest of the additional information will get inflated alongside.
4. `.series_plotter` should be designed analogous to the other plotting functions including the set parameter for colour_by, size_by and linetype_by as well overloading via `...` for adjusting the default values for size, linetype showing legends, etc.

Reply:
Now I have implemented .get_assay_data. 

But I have a problem with .incorporate_series_vis:

I get the feature_data, but where it should be stored? The feature_datas differs in length, so they can't be stored in same data frame just like that. 

Because the data is not melted, feature_datas can not be just assigned to xy-data points. 
I thought that I just take feature_datas as a list of lists, but I'm not sure if that is acceptable or smart thing to do.

Reply:
the feature data should have the same length as the chosen dimension. Beaware, that `retrieveCellInfo/FeatureInfo` return a named list: `value` is the actual data and `name` is the retrieved name. Due to the support for partial matching this is necessary.

For insparation look at other plotting functions in `miaViz`. For example in `plotGraph.R`:

```
 for(i in seq_along(variables)){
    # get data
    feature_info <- type_FUN(se, variables[i],
                             exprs_values = by_exprs_values)
    feature_info_name <- feature_info$name
    # mirror back variable name, if a partial match was used
    var_name <- names(variables)[i]
    assign(var_name, .get_new_var_name_value(get(var_name),
                                             feature_info$name))
    # rename columns by their usage
    feature_info$name <- var_name
    graph_data <- .add_graph_data_or_warn(feature_info, graph_data,
                                          type = "nodes",
                                          feature_info_name)
}
```

Reply:
Yes, of course... I was in the wrong track